src/crossroads/main.py

```diff
@@ -107,3 +137,48 @@ def json_parser():
                                      longitude=(location['longitudeE7'] / 10. ** 7)))
 
     return parsed_locations
+
+def add_color_marker(parsed_locations_list, colour):
+    """
+    Adds the "marker" key and value depending on the parameter "colour" to the parsed .json file with Location History.
+
+    :param parsed_locations_list: list of dictionaries
+    :param colour: (str) the color of the marker to be added
+    :return parsed_loc_list_with_colour: a list of chronologically arranged dictionaries, from the oldest ones
+    """
+    parsed_loc_list_with_colour = parsed_locations_list
+
+    if colour == 'blue':
+        marker = 'http://maps.google.com/mapfiles/ms/icons/blue-dot.png'
+    else:
+        marker = 'http://maps.google.com/mapfiles/ms/icons/red-dot.png'
+
+    for location in parsed_loc_list_with_colour:
+        parsed_loc_list_with_colour[parsed_loc_list_with_colour.index(location)]['icon'] = marker
+
+    return parsed_loc_list_with_colour
+
+def validate_json(json_file):
```

Comment:
This function does not accept json file, but json content. I propose to rename it and all related names in this function to `json_content`